install.sh

```diff
@@ -8,28 +8,30 @@
 # This script will create links for the dotfiles.
 
 # Step 1: ensure all necessary directories exist before linking files.
-for dir in $(find dotfiles/ -mindepth 1 -type d); do
+for dir in $(ls | find * -mindepth 1 -type d ! -name '.git' ! -name '.gitignore'); do
```

Comment:
The `find` command here only searches for directories (`-type d`).
You have added `! -name '.gitignore'`, which is always a file.
So either you need to remove the `-type d` (if you want to include files), or you need to remove the `! -name '.gitignore'` (because it has no effect if you are only finding directories).

Reply:
What if we negate the `.git` directory.